### ISSUE TITLE
Midi Support?

### ISSUE THREAD

**Issue by Cyantig** [zydeco/minivmac4ios#62]
Before installing I wanted to ask if the ios version of minivmac would be able to output midi data to ios? I have a project idea that involves midi data.

**Comment by zydeco** [comment]
No, Mini vMac doesn't emulate any devices attached to the serial ports, other than LocalTalk over UDP, which is not enabled on iOS due to system limitations on UDP usage.
I'm not familiar with how MIDI works at low level, but if the Mac apps are just sending raw MIDI packets through the serial port, it might be straight forward to implement.

What apps are you looking to use on the emulated Mac side?

**Comment by Cyantig** [comment]
> No, Mini vMac doesn't emulate any devices attached to the serial ports, other than LocalTalk over UDP, which is not enabled on iOS due to system limitations on UDP usage. I'm not familiar with how MIDI works at low level, but if the Mac apps are just sending raw MIDI packets through the serial port, it might be straight forward to implement.
> 
> What apps are you looking to use on the emulated Mac side?

Music Mouse for the MacPlus. It's an algorithmic musical composition software designed by Laurie Spiegel.

**Comment by brnrds** [comment]
[This](https://www.emaculation.com/forum/viewtopic.php?t=10420) works, both the provided build and the patch against the latest version of Mini vMac compiled for MacOS. I found it specifically while trying to get Music Mouse to send MIDI to a DAW in the host OS for Mini vMac and can confirm it's definitely possible.

That said, I'm trying to first get minivmac4ios to build for my Ipad 2, so that I can next try to use a MIDI patched version of Mini vMac to see if it will work, which I think is precisely what @Cyantig is trying to accomplish. Still stuck on the "getting minivmac4ios to build" bit, which is why I came across this issue, while looking for issues that might help. 

In short, Music Mouse can send MIDI via a patched Mini vMac to MacOS (and Ableton Live in my case) but I'm not sure if that patch will work with iOS.
